sys/fs/fuse/fuse_vnops.c

```diff
@@ -3201,24 +3197,6 @@ fuse_vnop_vptofh(struct vop_vptofh_args *ap)
 			"VOP_VPTOFH without FUSE_EXPORT_SUPPORT");
 		return EOPNOTSUPP;
 	}
-	if ((mp->mnt_flag & MNT_EXPORTED) &&
-		!(data->dataflags & FSESS_NO_OPENDIR_SUPPORT))
-	{
-		/*
-		 * NFS is stateless, so nfsd must reopen a directory on every
-		 * call to VOP_READDIR, passing in the d_off field from the
-		 * final dirent of the previous invocation.  But without
-		 * FUSE_NO_OPENDIR_SUPPORT, the FUSE protocol does not
-		 * guarantee that d_off will be valid after a directory is
-		 * closed and reopened.  So prohibit exporting FUSE file
-		 * systems that don't set that flag.
-		 *
-		 * But userspace NFS servers don't have this problem.
-                 */
-		SDT_PROBE2(fusefs, , vnops, trace, 1,
-			"VOP_VPTOFH without FUSE_NO_OPENDIR_SUPPORT");
-		return EOPNOTSUPP;
-	}
```

Comment:
The check is still important, even if the condition is not correct.  I think you should reinstate the check, but change the condition to `fsess_is_impl(mp, FUSE_OPENDIR)`.  If a fuse filesystem implements FUSE_OPENDIR, then it cannot be correctly exported via NFS.  Then you'll also need to add a `fsess_set_impl` to `fuse_filehandle_open`.

Finally, we should probably stop returning an error.  Instead, we should probably allow the operation to succeed, but use a `fuse_warn` to warn the sysadmin.  But that's beyond the scope of this PR.